server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -661,6 +661,13 @@ private void internalRecoverFromStore(IndexShard indexShard) throws IndexShardRe
                 indexShard.recoveryState().getIndex().setFileDetailsComplete();
             }
             indexShard.openEngineAndRecoverFromTranslog();
+            if (indexShard.shouldSeedRemoteStore()) {
+                indexShard.getThreadPool()
+                    .executor(ThreadPool.Names.GENERIC)
+                    .execute(() -> { indexShard.refresh("remote store migration"); });
+                indexShard.waitForRemoteStoreSync();
```

Comment:
what happens if recovery times out and fails?

Reply:
It will be retried and the upload will consider the delta of the files to be uploaded